backend/src/apiserver/config/config.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/kubeflow/pipelines/backend/src/apiserver/resource"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/server"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+	"google.golang.org/grpc/codes"
+	"os"
+	"time"
+)
+
+// deprecated
+type deprecatedConfig struct {
+	Name        string
+	Description string
+	File        string
+}
+
+type configPipelines struct {
+	Name        string
+	Description string
+	File        string
+	// optional, Name is used for PipelineVersion if not provided
+	VersionName string
+	// optional, Description is used for PipelineVersion if not provided
+	VersionDescription string
+}
+
+type config struct {
+	// If pipeline version already exists and
+	// LoadSamplesOnRestart is enabled, then the pipeline
+	// version is uploaded again on server restart
+	// if it does not already exist
+	LoadSamplesOnRestart bool
+	Pipelines            []configPipelines
+}
+
+// LoadSamples preloads a collection of pipeline samples
+//
+// If LoadSamplesOnRestart is false then Samples are only
+// loaded once when the pipeline system is initially installed.
+// They won't be loaded on upgrade or pod restart, to
+// prevent them from reappearing if user explicitly deletes the
+// samples. If LoadSamplesOnRestart is true then PipelineVersions
+// are uploaded if they do not already exist upon upgrade or pod
+// restart.
+func LoadSamples(resourceManager *resource.ResourceManager, sampleConfigPath string) error {
+	pathExists, err := client.PathExists(sampleConfigPath)
+	if err != nil {
+		return err
+	}
+
+	if !pathExists {
+		glog.Infof("No samples path provided, skipping loading samples..")
+		return nil
+	}
+
+	configBytes, err := os.ReadFile(sampleConfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to read sample configurations file. Err: %v", err)
+	}
+
+	var pipelineConfig config
+	if configErr := json.Unmarshal(configBytes, &pipelineConfig); configErr != nil {
+		// Attempt to parse to deprecated config version:
+		var deprecatedCfg []deprecatedConfig
+		if depConfigErr := json.Unmarshal(configBytes, &deprecatedCfg); depConfigErr != nil {
+			return fmt.Errorf("failed to read sample configurations. Err: %v", configErr)
+		}
+		glog.Warningf("encountered deprecated version of samples config, please update to the newer version to " +
+			"ensure future compatibility")
+		for _, cfg := range deprecatedCfg {
+			pipelineConfig.Pipelines = append(pipelineConfig.Pipelines, configPipelines{
+				Name:        cfg.Name,
+				File:        cfg.File,
+				Description: cfg.Description,
+			})
+		}
+		pipelineConfig.LoadSamplesOnRestart = false
+	}
+
+	// Check if sample has been loaded already and skip loading if true.
+	haveSamplesLoaded, err := resourceManager.HaveSamplesLoaded()
+	if err != nil {
+		return err
+	}
+
+	if !pipelineConfig.LoadSamplesOnRestart && haveSamplesLoaded {
+		glog.Infof("Samples already loaded in the past. Skip loading.")
+		return nil
+	}
```

Comment:
Optional:
```suggestion
	if !pipelineConfig.LoadSamplesOnRestart {
		// Check if sample has been loaded already and skip loading if true.
		haveSamplesLoaded, err := resourceManager.HaveSamplesLoaded()
		if err != nil {
			return err
		}

		if haveSamplesLoaded {
			glog.Infof("Samples already loaded in the past. Skip loading.")
			return nil
		}
	}
```

Reply:
note that we can't move scope for haveSamplesLoaded, we're utilizing it [here](https://github.com/kubeflow/pipelines/pull/11553/files#diff-56a9fe52a26fedc39a572ce772a5fbe999093e293eb4bd261ec69b60aafe6ed4R203), we can declare it outside, but that opens us for nil pointer concerns, so I prefer the current approach